py4DSTEM/process/ptychography/utils.py

```diff
@@ -3,12 +3,211 @@
+    + 1 / 4 * aberr.C32b * (2 * u3 * v + 2 * u * v3)
+
+    chi *= 2 * np.pi / lam
+
+    return chi
+
+def aperture3(qx, qy, lam, alpha_max):
+    """
+       Return a boolean where (qx,qy) is within a sharp aperture given wavelength and convergence semi-angle
+
+        Args:
+            qx (float): Qx components
+            qy (float, float, 1D): Qy components
+            lam (float): wavelength in angstrom
+            alpha_max (float): convergence semi-angle in rad
+        Returns:
+            bool, True if sqrt(qx**2 + qy**2) * lam < alpha_max
+    """
+
+    xp = sp.get_array_module(qx)
+    qx2 = qx ** 2
+    qy2 = qy ** 2
+    q = xp.sqrt(qx2 + qy2)
+    ktheta = xp.arcsin(q * lam)
+    return ktheta < alpha_max
+
+
+def aperture_xp(qx, qy, lam, alpha_max, edge=2):
+    """
+           Return a boolean where (qx,qy) is within a sharp aperture given wavelength and convergence semi-angle
+
+            Args:
+                qx (float): Qx components
+                qy (float, float, 1D): Qy components
+                lam (float): wavelength in angstrom
+                alpha_max (float): convergence semi-angle in rad
+            Returns:
+                bool, True if sqrt(qx**2 + qy**2) * lam < alpha_max
+    """
+
+    xp = sp.get_array_module(qx)
+    q = xp.sqrt(qx ** 2 + qy ** 2)
+    ktheta = xp.arcsin(q * lam)
+    qmax = alpha_max / lam
+    dk = qx[0][1]
+
+    arr = xp.zeros_like(qx)
+    arr[ktheta < alpha_max] = 1
+    # riplot(arr,'arr')
+    if edge > 0:
+        dEdge = edge / (qmax / dk);  # fraction of aperture radius that will be smoothed
+        # some fancy indexing: pull out array elements that are within
+        #    our smoothing edges
+        ind = (ktheta / alpha_max > (1 - dEdge)) * (ktheta / alpha_max < (1 + dEdge))
+        arr[ind] = 0.5 * (1 - xp.sin(np.pi / (2 * dEdge) * (ktheta[ind] / alpha_max - 1)))
+    return arr
+
+def fftshift_checkerboard(w, h):
+    """
+       Return a checkerboard array of size (w*2,h*2), where the values +1 and -1 are alternating.
+       This can be used to perform an fftshift operation in-place, when used in conjunction with a Fourier transform.
+
+        Args:
+            w (int): width of the array / 2
+            h (int): height of the array / 2
+        Returns:
+            checkerboard array of size (w*2,h*2)
+    """
+
+    re = np.r_[w * [-1, 1]]  # even-numbered rows
+    ro = np.r_[w * [1, -1]]  # odd-numbered rows
+    return np.row_stack(h * (re, ro))
+
+def fourier_coordinates_2D(N, dx=[1.0, 1.0], centered=True):
+    """
+        Generates Fourier coordinates for a (Nx,Ny)-shaped 2D array.
+    	Specifying the pixelSize argument sets a unit size.
+    """
+    qxx = np.fft.fftfreq(N[1], dx[1])
+    qyy = np.fft.fftfreq(N[0], dx[0])
+    if centered:
+        qxx += 0.5 / N[1] / dx[1]
+        qyy += 0.5 / N[0] / dx[0]
+    qx, qy = np.meshgrid(qxx, qyy)
+    q = np.array([qy, qx]).astype(np.float32)
+    return q
```

Comment:
so i guess we're still keeping this separate from the stuff in `process.utils` for now, and keeping `centered=True` as default, yes?  it would be nice to make these one fn at some point (i.e. this fn and `process.utils.make_Fourier_coords2D`), but it is not necessary now!